### DISCUSSION TITLE
datadir in k3s install script

### DISCUSSION THREAD

**Discussion by cwrx777** [k3s-io/k3s#9498] [Q&A]
Hi,

I notice in the [k3s install script](https://get.k3s.io/), there are references to hardcoded value of `/var/lib/rancher/k3s`. This directory is also the [root data directory for k3s server](https://docs.k3s.io/cli/server#data), which can be overridden by specifying `--data-dir` option when running the server. Does it make sense to allow overriding of `/var/lib/rancher/k3s` in install script? or is the directory in install script and in k3s server for different purpose?

**Comment by brandond** [comment]
The install script only works on the default paths, and does not consume the k3s config file or CLI args. If you customize the data-dir then you'll need to handle cleanup manually.

note that the references to /var/lib/rancher/ are only in the killall script which is created by the installer; we are working on rolling that into the main k3s binary at which point it will support custom data-dirs.
